processor/src/errors.rs

```diff
@@ -21,10 +29,99 @@ pub enum ExecutionError {
     NttDomainSizeTooSmall(u64),
     NttDomainSizeNotPowerof2(u64),
     InvalidStackDepthOnReturn(usize),
-    InterpolationResultSizeTooBig(u64),
+    InterpolationResultSizeTooBig(usize, usize),
     NotBinaryValue(Felt),
     NotU32Value(Felt),
     ProverError(ProverError),
     SyscallTargetNotInKernel(Digest),
     UnexecutableCodeBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
+        match self {
+            ExecutionError::AdviceKeyNotFound(key) => {
+                let hex = to_hex(Felt::elements_as_bytes(key))?;
+                write!(fmt, "Can't write to advice tape: value for key {hex} not present in the advice map.")
+            }
+            ExecutionError::AdviceSetLookupFailed(reason) => {
+                write!(fmt, "Advice set lookup failed: {reason}")
+            }
+            ExecutionError::AdviceSetNotFound(root) => {
+                write!(fmt, "Advice set with root {root:x?} not found")
+            }
+            ExecutionError::AdviceSetUpdateFailed(reason) => {
+                write!(fmt, "Advice set update failed: {reason}")
+            }
+            ExecutionError::AdviceTapeReadFailed(step) => {
+                write!(fmt, "Advice tape read failed at step {step}")
+            }
```

Comment:
nit: I think we can improve readability here by importing `ExecutionError` variants. For example, this could look like so:
```Rust
fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
    use ExecutionError::*;
    match self {
        AdviceKeyNotFound(key) => {
            let hex = to_hex(Felt::elements_as_bytes(key))?;
            write!(fmt, "Can't write to advice tape: value for key {hex} not present in the advice map.")
        }
        AdviceSetLookupFailed(reason) => write!(fmt, "Advice set lookup failed: {reason}"),
        AdviceSetNotFound(root) => write!(fmt, "Advice set with root {root:x?} not found"),
        AdviceSetUpdateFailed(reason) => write!(fmt, "Advice set update failed: {reason}"),
        AdviceTapeReadFailed(step) => write!(fmt, "Advice tape read failed at step {step}"),
       ...
```

Reply:
`rustfmt` is automatically adding the brackets.

Reply:
maybe this option does what you suggested? 

https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=match#match_arm_blocks

Reply:
ah, my bad, it doesn't add the brackets when removing `ExecutionError::`. sorry! Now it is changed.